### PR TITLE
Return percentage error

### PR DESCRIPTION
**Pull request by aymerickmichelet**

### CODE REVIEW

**Review by aymerickmichelet — COMMENTED** [review]
I will implement other feat (like return csv_error path) later ^^
But now, I want to implement pandas branch! So, I need to merge on master

File: app.py [hunk 1]
```diff
@@ -45,16 +50,14 @@ def parse_csv():
...
 
+    results = {
+        "correct": '{percent:.2%}'.format(percent=0.6686),
+        "warning": '{percent:.2%}'.format(percent=0.1391),
+        "error": '{percent:.2%}'.format(percent=0.1923)
+    }
 
-if __name__ == '__main__':
-    app.run(debug=True)
+    return render_template("result.html", results=results)
```
Comment (TheJulienM) [inline app.py:60]: Why is that?

File: templates/index.html [hunk 2]
```diff
@@ -14,9 +14,7 @@ <h3>Give your csv, we say you if you have wrong data.</h3>
             <div class="outer_quare">
                 <label id="inner_square" class="pointer inner_square" for="upload">
                     <div id="loader" class="hide lds-ring"><div></div><div></div><div></div><div></div></div>
-
                         <input id="upload" name="upload" class="upload" type="file" accept=".csv"/>
-
                     <img id="csv_icon" class="csv_icon" src="../static/csv.svg" alt="svg file logo" />
                     <span id="label" class="label">Click here to upload your csv file</span>
```
Comment (TheJulienM) [inline templates/index.html:19]: It must be "xlsx" now, right?

File: templates/result.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <link href="{{ url_for('static',filename='style_result.css') }}" rel="stylesheet">
+        <link rel="icon" href="{{ url_for('static',filename='csv.svg') }}" />
+        <title>Data quality</title>
+        <script src="https://kit.fontawesome.com/e36d430f89.js" crossorigin="anonymous"></script>
+    </head>
+    <body>
+        <h1>Csv data quality</h1>
```
Comment (TheJulienM) [inline templates/result.html:12]: Same here no ?

File: templates/result.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                    <span class="percentage_label">usable data</span>
+                </div>
+                <div class="inner_inner_square_result red">
+                    <span class="percentage_number white">{{ results.error }}</span>
+                    <span class="percentage_label">unusable data</span>
+                </div>
+            </div>
+        </div>
+        <a class="download_correction white pointer" href="../static/csv.svg" download>
+            lenomducsv.csv
```
Comment (TheJulienM) [inline templates/result.html:31]: Change button text to "Download CSV File" ?

**Comment by aymerickmichelet** [comment]
Okay I will apply all your request

File: app.py (hunk 1 above)
Comment (aymerickmichelet) [reply app.py:60]: Yes I don't understand this display but please check in the [full file](https://github.com/aymerickmichelet/data_quality/blob/070715099f2ec70a43588c4546c126502195da30/app.py)
Normally, It's okay

Comment (TheJulienM) [reply app.py:60]: Weird but why not

File: app.py [hunk 5]
```diff
@@ -21,6 +21,11 @@ def hello_world():
     return render_template("index.html")
 
 
+@app.route('/result')
+def result():
+    return render_template("result.html")
```
Comment (TheJulienM) [inline app.py:26]: No error ? The template is waiting for an "results" array

Comment (aymerickmichelet) [reply app.py:25]: I will check

Comment (aymerickmichelet) [reply app.py]: indeed, fix

**Review by aymerickmichelet — COMMENTED** [review]
I think it's good now ^^

File: app.py [hunk 6]
```diff
@@ -13,16 +13,15 @@
...
 
 
 @app.route('/')
 def hello_world():
     return render_template("index.html")
 
 
-@app.route('/parse-csv', methods=['POST', 'GET'])
-def parse_csv():
+@app.route('/parse-xlsx', methods=['POST', 'GET'])
```
Comment (TheJulienM) [inline app.py:23]: Good idea !

File: service/uploader.py [hunk 7]
```diff
@@ -3,7 +3,7 @@
 import uuid
 import os
 
-CSV_EXTENSION = ".csv"
+CSV_EXTENSION = ".xlsx"
```
Comment (TheJulienM) [inline service/uploader.py:6]: You should rename the var name 😅 (I'm boring I know)

Comment (aymerickmichelet) [reply service/uploader.py:6]: Yes it's true ^^

**Pull request merged**